webpack/redux/reducers/RedHatRepositories/enabled.js

```diff
@@ -58,39 +58,47 @@ const changeRepoState = (state, repoToChange, stateDiff) => (
 );
 
 export default (state = initialState, action) => {
-  if (action.type === DISABLE_REPOSITORY_REQUEST) {
-    return changeRepoState(state, action.repository, { loading: true });
-  } else if (action.type === DISABLE_REPOSITORY_FAILURE) {
-    return changeRepoState(state, action.payload.repository, { loading: false });
-  } else if (action.type === ENABLED_REPOSITORIES_REQUEST) {
-    if (!action.silent) {
-      return state.set(['loading'], true);
+  switch (action.type) {
+    case DISABLE_REPOSITORY_REQUEST:
+      return changeRepoState(state, action.repository, { loading: true });
+    case DISABLE_REPOSITORY_FAILURE:
+      return changeRepoState(state, action.payload.repository, { loading: false });
+    case ENABLED_REPOSITORIES_REQUEST:
+      if (!action.silent) {
+        return state.set(['loading'], true);
+      }
+      return state;
+    case ENABLED_REPOSITORIES_SUCCESS: {
+      const {
+        page,
+        per_page, // eslint-disable-line camelcase
+        subtotal,
+        results,
+      } = action.response;
+      return Immutable({
+        repositories: mapRepositories(results),
+        pagination: {
+          page: Number(page) || 1,
+          // server can return per_page: null when there's error in the search query,
+          // don't store it in such case
+          // eslint-disable-next-line camelcase
+          perPage: Number(per_page || state.pagination.perPage),
+        },
+        itemCount: Number(subtotal),
+        loading: false,
+        searchIsActive: !isEmpty(action.search),
+        search: action.search,
+      });
     }
-  } else if (action.type === ENABLED_REPOSITORIES_SUCCESS) {
-    const {
-      page, per_page, subtotal, results, // eslint-disable-line camelcase
-    } = action.response;
-
-    return Immutable({
-      repositories: mapRepositories(results),
-      pagination: {
-        page: Number(page) || 1,
-        // server can return per_page: null when there's error in the search query,
-        // don't store it in such case
-        // eslint-disable-next-line camelcase
-        perPage: Number(per_page || state.pagination.perPage),
-      },
-      itemCount: Number(subtotal),
-      loading: false,
-      searchIsActive: !isEmpty(action.search),
-      search: action.search,
-    });
-  } else if (action.type === ENABLED_REPOSITORIES_FAILURE) {
-    return Immutable({
-      repositories: [],
-      loading: false,
-      error: action.error,
-    });
+    case ENABLED_REPOSITORIES_FAILURE: {
+      return Immutable({
+        repositories: [],
+        loading: false,
+        error: get(action, ['error']),
+        missingPermissions: get(action, ['result', 'response', 'data', 'error', 'missing_permissions']),
```

Comment:
`lodash.get` is usually used to set up a default value when undefined is returned from the entered path,
For this case I would use the regular destructing. If it's not too difficult there's already the action destructing above
which can be reused.

Reply:
another thought - in my opinion the actions is the place to write "dirty" code that do all of the adjustments,
and the reducer should receive it as clean as possible.

Reply:
@laviro good point, the latter makes sense to me! I moved the "dirty" data traversal to the action creator.

Regarding `lodash.get`, In addition to providing default values, `get` also guards against throwing errors if one of your intermediary keys is undefined (for example, `result` or `data` etc.)
After moving it to the action, I tried to use destructuring to get the value, and I had something like
```
const { response: { data: { error: { missing_permissions } } } } = error
```
But it broke another test.  So I switched back to using `get`.
I know I could still use destructuring by using default values, like 
```
const { response: { data: { error: { missing_permissions } =  {}} =  {}} = error
```
But I feel like this is hard to understand and easy to get the syntax wrong, whereas lodash `get` is very straightforward.

Until JS optional chaining becomes part of JavaScript, I feel like lodash `get` is a good solution.